walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,18 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Arrays;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    //throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (events.isEmpty()) {
+        return Arrays.asList(TimeRange.WHOLE_DAY);
+    }
+
+    if (request.getDuration() > (long)TimeRange.WHOLE_DAY.duration()) {
+        return Arrays.asList();
+    }
+    return null;
```

Comment:
what would say is the difference between returning null or an empty list here? How would the caller of the method account for a null result or an empty list result?

Reply:
Returning null is not part of the method, I just placed it as a placeholder return so the function would have a reachable return statement in the meantime. The reason I return an empty list is because the class that is testing this code expects an empty list if an event whose duration is longer than a day is attempted to be scheduled.